src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddCaseNoteCallbackHandlerTest.java

```diff
@@ -0,0 +1,130 @@
+import static uk.gov.hmcts.reform.civil.utils.ElementUtils.unwrapElements;
+import static uk.gov.hmcts.reform.civil.utils.ElementUtils.wrapElements;
+
+@SpringBootTest(classes = {
+    AddCaseNoteCallbackHandler.class,
+    JacksonAutoConfiguration.class,
+    ValidationAutoConfiguration.class,
+    CaseNoteService.class,
+    CaseDetailsConverter.class
+})
+
+class AddCaseNoteCallbackHandlerTest extends BaseCallbackHandlerTest {
+
+    @Autowired
+    private AddCaseNoteCallbackHandler handler;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private CaseDetailsConverter caseDetailsConverter;
+
+    @MockBean
+    private IdamClient idamClient;
+
+    @Nested
+    class AboutToStartCallback {
+
+        @Test
+        void shouldReturnNoError_WhenAboutToStartIsInvoked() {
+            CaseDetails caseDetails = CaseDetailsBuilder.builder().atStateCaseIssued().build();
+            CallbackParams params = CallbackParamsBuilder.builder().of(ABOUT_TO_START, caseDetails).build();
+
+            AboutToStartOrSubmitCallbackResponse response = (AboutToStartOrSubmitCallbackResponse) handler
+                .handle(params);
+
+            assertThat(response.getErrors()).isNull();
+        }
+    }
+
+    @Nested
+    class AboutToSubmit {
+
+        @BeforeEach
+        void setup() {
+            given(idamClient.getUserDetails(any())).willReturn(UserDetails.builder()
+                                                                   .forename("John")
+                                                                   .surname("Smith")
+                                                                   .build());
+        }
+
+        private CaseNote caseNote(LocalDate date, String createdBy, String note) {
+            return CaseNote.builder()
+                .date(date)
+                .createdBy(createdBy)
+                .note(note)
+                .build();
+        }
+
+        @Test
+        void shouldAddCaseNoteToList_whenInvoked() {
+
+            CaseNote caseNote = caseNote(LocalDate.of(2021, 7, 5), "John Doe", "Existing note");
+
+            CaseDetails caseDetails = CaseDetails.builder()
+                .data(Map.of(
+                    "caseNote", "Example case note",
+                    "caseNotes", wrapElements(caseNote)))
+                .build();
+            CaseData data = caseDetailsConverter.toCaseData(caseDetails);
+
+            CallbackParams params = CallbackParamsBuilder.builder().of(ABOUT_TO_SUBMIT, data).build();
+
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+            CaseData updatedData = objectMapper.convertValue(response.getData(), CaseData.class);
+
+            assertThat(updatedData.getCaseNote()).isNull();
+            assertThat(unwrapElements(updatedData.getCaseNotes()))
+                .containsExactly(caseNote, caseNote(LocalDate.now(), "John Smith", "Example case note"));
```

Comment:
When we mock `CaseNoteService`, we could use `verify` to check if both `buildCaseNote` and `addNoteToList` methods were called.